Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -306,6 +327,40 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
         }
 
         /// <summary>
+        /// <para> Return true if any of below cases is satisfied:</para>
+        /// <para>      1) Package with same version and from same feed already exist in local repo</para>
+        /// <para>      2) If given feedUrl is null</para>
+        /// <para>              Package with same version from default feed already exist in local repo</para>
+        /// <para>      3) If given version and feedUrl are null</para>
+        /// <para>              Same version after query default feed for latest package</para>
```

Comment:
i think i am missing one more condition.

if feed is null, i should try the feed from the installed package, only if the feed from local package is also null, then fallback to default feed